src/connection/index.ts

```diff
@@ -136,7 +136,7 @@ export class ConnectionImpl implements Connection {
    * Remove stream registry after it is closed
    */
   removeStream (id: string) {
-
+    this.streams.splice(this.streams.findIndex(s => s.id === id), 1)
```

Comment:
This needs to guard on `findIndex` returning `-1`!

```js
[0, 1, 2].splice(-1)
// [2]
```

A test would help here too.

Reply:
Good catch! Thanks